util/memory/tracker.go

```diff
@@ -165,7 +165,11 @@ func (t *Tracker) remove(oldChild *Tracker) {
 			continue
 		}
 
-		t.Consume(-oldChild.BytesConsumed())
+		func() {
+			t.mu.Unlock()
+			defer t.mu.Lock()
+			t.Consume(-oldChild.BytesConsumed())
+		}()
```

Comment:
How about these implementation?
```go
func (t *Tracker) remove(oldChild *Tracker) {
	found := false
	t.mu.Lock()
	for i, child := range t.mu.children {
		if child == oldChild {
			t.mu.children = append(t.mu.children[:i], t.mu.children[i+1:]...)
			found = true
		}
	}
	t.mu.Unlock()
	if found {
		oldChild.setParent(nil)
		t.Consume(-oldChild.BytesConsumed())
	}
}
```

Reply:
Great. That is so neat. I've changed the code accordingly.